src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/ClientCertificateStore.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.certificate.infra;
+
+import com.aws.greengrass.clientdevices.auth.ClientDevicesAuthService;
+import com.aws.greengrass.clientdevices.auth.certificate.CertificateHelper;
+import com.aws.greengrass.clientdevices.auth.configuration.RuntimeConfiguration;
+import com.aws.greengrass.lifecyclemanager.Kernel;
+import com.aws.greengrass.util.FileSystemPermission;
+import com.aws.greengrass.util.platforms.Platform;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Optional;
+import javax.inject.Inject;
+
+/**
+ * A KeyStore for client certificates. When a client device provides their certificate we store it
+ * in this store it so that later we can refresh it later on using the cloud API.
+ */
+public class ClientCertificateStore {
+    private static final String STORE_NAME = "client.jks";
+    private final Path storePath;
+    private final RuntimeConfiguration runtimeConfiguration;
+    private KeyStore keyStore;
+    private char[] passphrase;
```

Comment:
My initial thought is that we can get away without a passphrase. It's not actually improving security and certificates are public information, anyway.

Reply:
Yeah I can remove that